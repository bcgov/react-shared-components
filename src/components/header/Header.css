.navbar > .container-fluid {
  width: 95%;
  justify-content: inherit;
}

.bcgov-pointer {
  cursor: pointer;
}
