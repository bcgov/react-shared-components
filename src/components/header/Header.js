/* eslint-disable no-alert */
import React from "react";
import PropTypes from "prop-types";
import "./Header.css";

const bcidSymbol = `${process.env.PUBLIC_URL}/images/bcid-symbol-rev.svg`;
const bcidLogoRev = `${process.env.PUBLIC_URL}/images/bcid-logo-rev-en.svg`;

export const HeadingTitle = (classNames) => (
  <div className={classNames} aria-labelledby="title" />
);

export const HeaderImage = (history, classNames, width, src) => (
  <img
    className={classNames}
    src={src}
    width={width}
    height="44"
    alt="B.C. Government Logo"
    onClick={() => history.push("/")}
    onKeyDown={() => history.push("/")}
    role="presentation"
  />
);

export const Header = ({ header: { name, history } }) => {
  return (
    <header>
      <nav className="container-fluid navbar navbar-expand-lg navbar-dark">
        {HeadingTitle("navbar-brand pointer")}
        {HeaderImage(
          history,
          "img-fluid d-none d-md-block pointer",
          181,
          bcidLogoRev
        )}
        {HeaderImage(history, "img-fluid d-md-none pointer", 64, bcidSymbol)}
        {HeadingTitle("pointer navbar-brand nav-item nav-link")}
        <div id="title" className="navbar-brand">
          {name}
        </div>
      </nav>
    </header>
  );
};

Header.propTypes = {
  header: PropTypes.shape({
    name: PropTypes.string.isRequired,
    history: PropTypes.any.isRequired,
  }).isRequired,
};
