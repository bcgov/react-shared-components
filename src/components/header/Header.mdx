import { Meta, Story } from "@storybook/addon-docs";
import { Header } from "./Header";

<Meta title="Header" component={Header} />

# Header

Headers help people understand what the product or service is about while providing a consistent
look, feel, and functionality across government sites.

## When to Use

This header must appear on all public-facing online B.C. Government content and services

## When not to Use

When the product or service is not owned by the Government of British Columbia

## API

| Prop   | Type   | Required | Description                              | Default |
| ------ | ------ | -------- | ---------------------------------------- | ------- |
| header | object | true     | object with details regarding the header |         |

### Header API

The header object has its own properties that need to be present within it, as shown below:

| Prop    | Type   | Required | Description                   |
| ------- | ------ | -------- | ----------------------------- |
| name    | string | true     | label for the header          |
| history | object | true     | javascript DOM history object |

<br />

## Basic Header

Used at the top of the page.

<Story name="Basic Header" id="header--basic"></Story>

```bash
import { useHistory } from "react-router-dom";

const header = {
  name: "Header title",
  history: useHistory()
};

<Header header={header} />;
```
