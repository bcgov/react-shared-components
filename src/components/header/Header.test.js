import React from "react";
import { createMemoryHistory } from "history";
import { render, fireEvent, getAllByAltText } from "@testing-library/react";
import testBasicSnapshot from "../../TestHelper";
import { Header, HeadingTitle, HeaderImage } from "./Header";

const clickHeaderAndCheckUpdatedHistory = (header, container, type) => {
  header.history.replace("/somepageroute");

  if (type === "click")
    fireEvent.click(getAllByAltText(container, "B.C. Government Logo")[0]);
  else if (type === "keydown")
    fireEvent.keyDown(getAllByAltText(container, "B.C. Government Logo")[0]);

  expect(header.history.location.pathname).toEqual("/");
};

describe("Header Component", () => {
  const header = {
    name: "File Submission",
    history: createMemoryHistory(),
  };

  const headerComponent = <Header header={header} />;

  test("Header matches the snapshot", () => {
    testBasicSnapshot(headerComponent);
  });

  test("HeadingTitle matches the snapshot", () => {
    const headingTitle = HeadingTitle("navbar-brand bcgov-pointer");

    testBasicSnapshot(headingTitle);
  });

  test("HeaderImage matches the snapshot", () => {
    const headingImage = HeaderImage(
      header.history,
      "img-fluid d-none d-md-block",
      181,
      "bcid-logo-rev-en.svg"
    );

    testBasicSnapshot(headingImage);
  });

  test("Clicking HeadingImage takes you back to home", () => {
    const { container } = render(headerComponent);

    clickHeaderAndCheckUpdatedHistory(header, container, "click");
  });

  test("Keydown on HeadingImage takes you back to home", () => {
    const { container } = render(headerComponent);

    clickHeaderAndCheckUpdatedHistory(header, container, "keydown");
  });
});
