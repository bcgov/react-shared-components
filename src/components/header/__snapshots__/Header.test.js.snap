// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header matches the snapshot 1`] = `
<header
  className="bcgov-header"
>
  <nav
    aria-label="Header"
    className="container-fluid navbar navbar-expand-lg navbar-dark"
  >
    <div
      aria-labelledby="title"
      className="navbar-brand bcgov-pointer"
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-none d-md-block bcgov-pointer"
      height="44"
      onClick={[Function]}
      onKeyDown={[Function]}
      src="undefined/images/bcid-logo-rev-en.svg"
      width={181}
    />
    <img
      alt="B.C. Government Logo"
      className="img-fluid d-md-none bcgov-pointer"
      height="44"
      onClick={[Function]}
      onKeyDown={[Function]}
      src="undefined/images/bcid-symbol-rev.svg"
      width={64}
    />
    <div
      aria-labelledby="title"
      className="bcgov-pointer navbar-brand nav-item nav-link"
    />
    <div
      className="navbar-brand"
      id="title"
    >
      File Submission
    </div>
  </nav>
</header>
`;

exports[`Header Component HeaderImage matches the snapshot 1`] = `
<img
  alt="B.C. Government Logo"
  className="img-fluid d-none d-md-block"
  height="44"
  onClick={[Function]}
  onKeyDown={[Function]}
  src="bcid-logo-rev-en.svg"
  width={181}
/>
`;

exports[`Header Component HeadingTitle matches the snapshot 1`] = `
<div
  aria-labelledby="title"
  className="navbar-brand bcgov-pointer"
/>
`;
